### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by BMateHU**

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -12,6 +12,8 @@ public class TorpedoStore {
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
 
+  private Random generator = new Random();
+
```
Comment (YinnemJun) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: nagyon jo

**Review by YinnemJun — CHANGES_REQUESTED** [review]
remove empty line at the end

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -14,6 +14,8 @@ public class TorpedoStore {
 
   private int torpedoCount = 0;
 
+  private Random generator = new Random();
```
Comment (Burrey) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: Beautiful line. Quite exquisite!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (Burrey) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: What an amazing fix! I commend!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
```
Comment (Burrey) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:39]: Bad line removed!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 5]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (Burrey) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: Exactly!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 6]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     }
 
     boolean success = false;
```
Comment (Burrey) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:39]: Remove this line!

**Review by YinnemJun — APPROVED** [review]

**Review by Burrey — APPROVED** [review]

**Pull request merged**